tests/macsec/test_interop_protocol.py

```diff
@@ -132,11 +133,33 @@ def test_snmp(self, duthost, ctrl_links, upstream_links, creds, wait_mka_establi
                     commands=['show snmp community | include name'])
                 community = re.search(r'Community name: (\S+)',
                                       result['stdout'][0]).groups()[0]
+                up_link = upstream_links[ctrl_port]
+                nbr_ip = up_link["local_ipv4_addr"]
             else:  # vsonic neighbour
                 community = creds['snmp_rocommunity']
+                # Use LoopbackIP to query from vsonic neighbor
+                result = nbr["host"].config_facts(host=nbr["host"].hostname,
+                                                  source="running")[
+                                                  "ansible_facts"].get(
+                                                  "LOOPBACK_INTERFACE",
+                                                  {}).get('Loopback0', {})
+                for ip in result:
```

Comment:
>for [](http://example.com/codeflow?start=16&length=3)

You may use `for-else` to capture the situation that no ip satisfy the condition. In the `else` block, you can explicitly fail the test.

Reply:
Fixed